purchase_procurement_analytic/models/stock.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.model
+    def _prepare_procurement_from_move(self):
+        res = super(StockMove, self)._prepare_procurement_from_move()
+        res['account_analytic_id'] = self.procurement_id.account_analytic_id.id
```

Comment:
What does this makes? Here you are preparing the procurement, but you reference the field that will host this created procurement, isn't it?

Reply:
I'll correct myself.

This code is necessary to propagate account analytic account from sale to purchase through procurement.

I've added again to make module work as we expect.